### ISSUE TITLE
Example usage: 

### ISSUE THREAD

**Issue by adibender** [nhejazi/haldensify#37]
Hi, this is regarding the software review here: https://github.com/openjournals/joss-reviews/issues/4522#4522

I like the vignette, but I think it could be improved a bit

- results plot should be described better. What do the colors indicate? What do we see? A legend that explains colors + dashed/solid lines etc. would probably be helpful
- would be nice to also see this applied to a real world example

**Comment by adibender** [comment]
Also, I ran the code and got the following output for the haldensify fit: 

```
HAL Conditional Density Estimation
Number of bins over support of A: 30
CV-selected lambda: 4e-04
Summary of fitted HAL:
         coef                term
 1:  3.615390         (Intercept)
 2: -7.698355  [ I(W >= -0.102) ]
 3: -7.138447  [ I(W >= -0.402) ]
 4: -7.086183  [ I(W >= -2.771) ]
 5: -6.995051  [ I(W >= -3.578) ]
 6:  6.074327 [ I(bin_id >= 15) ]
 7: -6.008023  [ I(W >= -0.642) ]
 8: -5.901929  [ I(W >= -2.361) ]
 9:  5.856157  [ I(bin_id >= 9) ]
10:  5.763998   [ I(W >= 0.683) ]
Warning message:
In summary.hal9001(x$hal_fit) : Coefficients for many lambda exist --
Summarizing coefficients corresponding to minimum lambda.
```

What does the warning mean? Should I be worried? Readers/Users of the vignette will probably have similar questions. Maybe address it in the vignette.

**Comment by nhejazi** [comment]
Thank you for your thorough review of this package @adibender! I've addressed the points above in https://github.com/nhejazi/haldensify/pull/38. The changes are summarized below

- I've added a paragraph below the plot in the vignette depicting the conditional density predictions against the theoretically correct densities for each of the groups. I found prose to be easier in getting the point across than a legend, the requisite concision of which could still omit important details.
- The major real-world use-case of this software is, as mentioned, for estimating the generalized propensity score in causal inference problems with continuous treatment variables. To that end, I've added a sentence saying that it is used in most applications of our 2020 _Biometrics_ paper. It is additionally used in the 2022 preprint described in the next section of the vignette.
- The warning is not a problem and, as indicated, is generated by `summary.hal9001`, a summary method called by `summary.haldensify`. I've suppressed this warning now in `summary.haldensify` as explaining this to a new user would be challenging, given that it requires a technical understanding of this algorithm and that implemented in the `hal9001` package.

**Comment by adibender** [comment]
Hi, thanks for the updates. 
Some additional minor points. 

- I've noticed you also added a section on IPW estimation. That's great, but makes the vignette somewhat lengthy. If it makes sense, consider splitting the vignette in two vignettes 
- The references don't seem to render correctly in some cases? In some cases there's just a _____ in the beginning. 
- Thanks for the references to the papers. In my experience for people interested in the software it's still helpful to see a concise walk through a real life example to get understanding of limitations, interpretation, etc. in absence of ground truth. But up to you.

**Comment by nhejazi** [comment]
thanks for the additional comments, a few responses and rationalizations below:

* I've noticed you also added a section on IPW estimation. That's great, but makes the vignette somewhat lengthy. If it makes sense, consider splitting the vignette in two vignettes
> The section on IPW estimation was actually already there, and, while I agree that the vignette is lengthier as a result, I'd rather have these two disjoint pieces presented together for a couple reasons. 1) A standalone vignette on IPW estimation would be quite short and would require the conditional density estimation material as prerequisite, so why not present it immediately after. 2) A user-side issue I've noticed with multiple vignettes is that users often only consult the first, so this would risk the IPW material being missed entirely. I'll keep this in mind and migrate the IPW material to a new section if I decide to expand this section significantly, which may yet happen since the statistical methodology paper on this is still only a preprint.

* The references don't seem to render correctly in some cases? In some cases there's just a _____ in the beginning.
> To the best of my knowledge, this is a stylistic choice made by the bibliographic management system that `pkgdown` uses under the hood (I'm not sure where the actual `.cls` file lives), but the underline/underscore is merely shorthand for indicating that the authorship team of the given article is the same as the preceding article. For example, you'll notice that in the rendered `README.md` (`index.html`), entries 5 and 6 in the bibliography display this, but this is because they are from the same set of authors as entry 4.

* Thanks for the references to the papers. In my experience for people interested in the software it's still helpful to see a concise walk through a real life example to get understanding of limitations, interpretation, etc. in absence of ground truth. But up to you.
> I agree with you here but feel that the vignette may be the wrong or a suboptimal place to present this information. My take is that it's much more useful to see real-world, possibly published examples of the methodology successfully applied, rather than including examples here for which context might be sparse. With coauthors, I've also been writing a to-be-published book that covers some of this material, with a section on conditional density estimation (https://tlverse.org/tlverse-handbook/sl3.html#pooled-hazard-regression), and I may add some more material there and link back here once we make further progress on that.

I appreciate your taking the time to review this thoroughly and will consider these finer points in improving the software package in the future.
